File: Mixter/Domain/UserFollowed.cs
namespace Mixter.Domain
{
    public struct UserFollowed : IDomainEvent
    {
        public SubscriptionId SubscriptionId { get; private set; }

        public UserFollowed(SubscriptionId subscriptionId)
            : this()
        {
            SubscriptionId = subscriptionId;
        }
    }
}

File: domain/src/main/scala/mixter/domain/subscription/event/UserFollowed.scala
package mixter.domain.subscription.event

import mixter.domain.subscription.SubscriptionId
import mixter.domain.{AggregateId, Event}

sealed trait SubscriptionEvent extends Event
case class UserFollowed(subscriptionId: SubscriptionId) extends SubscriptionEvent{
  override def id: AggregateId = subscriptionId
}
